inductiva/fluids/scenarios/coastal_area/bathymetry.py

```diff
@@ -0,0 +1,223 @@
+              water. Must be between 0 and 100.
+        """
+
+        corner_values = [
+            random.uniform(0, max_depth),
+            random.uniform(0, max_depth),
+            random.uniform(-max_depth, 0),
+            random.uniform(-max_depth, 0),
+        ]
+
+        depths = inductiva.generative.procedural.generate_random_map_level(
+            x_num, y_num, corner_values, initial_roughness, roughness_factor)
+
+        depths = inductiva.generative.procedural.adjust_map_level(
+            depths, percentile_above_water)
+
+        x, y = np.meshgrid(np.linspace(*x_range, x_num),
+                           np.linspace(*y_range, y_num),
+                           indexing="ij")
+
+        return cls(depths.flatten(), x.flatten(), y.flatten())
+
+    def to_text_file(self, text_file_path: str):
+        """Writes the bathymetry to a text file.
+
+        Args:
+            text_file_path: Path to the text file.
+        """
+
+        np.savetxt(text_file_path, self.depths)
+
+    @property
+    def x_range(self) -> Tuple[float]:
+        """Returns the range of x values."""
+
+        return (np.min(self.x), np.max(self.x))
+
+    @property
+    def y_range(self) -> Tuple[float]:
+        """Returns the range of y values."""
+
+        return (np.min(self.y), np.max(self.y))
+
+    def x_uniques(self, sort: bool = False) -> np.ndarray:
+        """Returns the unique x values.
+        
+        Args:
+            sort: Whether to sort the unique values.
+        """
+        x_uniques = np.unique(self.x)
+        if sort:
+            x_uniques = np.sort(x_uniques)
+        return x_uniques
+
+    def y_uniques(self, sort: bool = False) -> np.ndarray:
+        """Returns the unique y values.
+        
+        Args:
+            sort: Whether to sort the unique values.
+        """
+        y_uniques = np.unique(self.y)
+        if sort:
+            y_uniques = np.sort(y_uniques)
+        return y_uniques
+
+    def is_uniform_grid(self) -> bool:
+        """Determines whether the bathymetry is defined on a uniform grid."""
+
+        x_uniques_diffs = np.diff(self.x_uniques(sort=True))
+        y_uniques_diffs = np.diff(self.y_uniques(sort=True))
+
+        return (np.unique(x_uniques_diffs.round(decimals=2)).size == 1 and
+                np.unique(y_uniques_diffs.round(decimals=2)).size == 1)
+
+    def plot(
+        self,
+        cmap: Optional[str] = None,
+        clim: Optional[Tuple[float]] = None,
+        path: Optional[str] = None,
+    ) -> Union[matplotlib.axes.Axes, None]:
+        """Plots the bathymetry.
+
+        The bathymetry is represented as a 2D map of depths, with the x and y
+        coordinates of the points where the depths are defined in the axes.
+
+        The plot is produced with matplotlib.
+    
+        Args:
+            cmap: Colormap to use. Defaults to the matplotlib default colormap.
+            clim: Color limits.
```

Comment:
Maybe explain this better, the range of values you want to plot. Btw, why did you expose it to the user? Is there interest to show values in other ranges other than (min, max) ?

Reply:
I'll add more info.

I think this parameter is useful to make sure that colors are the same in different plots, so I tend to add to all plots of this kind. Users will let us know, anyway :)